### DISCUSSION TITLE
Unable to setState or use selectedNodes

### DISCUSSION THREAD

**Discussion by SrivyshnaviK** [dowjones/react-dropdown-tree-select#541] [Q&A]
Any code injection into onChange function like setting state using hook or dispatch checkbox functionality is not working.

**Comment by mrchief** [comment]
> Any code injection into onChange function like setting state using hook or dispatch checkbox functionality is not working.

What do you mean?

**Comment by SrivyshnaviK** [comment]
I tried to use those selectedNodes by injecting them into a state like
setData(selectedNodes).. This call is interrupting check box functionality.

**Comment by mrchief** [comment]
So you're most likely re-rendering the component. The component maintains its own state so you can take a look at https://github.com/dowjones/react-dropdown-tree-select/blob/develop/docs/HOC.md#prevent-re-render-on-parent-propstate-changes approach. If you search the old issues, there's a hooks based implementation as well.
